src/main/java/org/magicdgs/thaplv/utils/stats/LengthBinning.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2016, Daniel Gomez-Sanchez <daniel.gomez.sanchez@hotmail> All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ * and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.utils.stats;
+
+import java.text.DecimalFormat;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+/**
+ * Class that bin a double based on the length, computing mean, median and quantiles.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+// TODO: implement unit tests
+public class LengthBinning {
+
+    // value that contains the bins
+    // TODO: implement a getter?
+    private final ConcurrentSkipListMap<Integer, RunningStats> binStat;
+
+    // value that contains the bin size
+    private final int binSize;
+
+    // which quantiles should be computed?
+    private final double[] quantiles;
+
+    private boolean empty;
+
+    /**
+     * Constructs a binning by length using the default quantiles (1, 5, 95, 99).
+     *
+     * @param binSize size of the bin.
+     */
+    public LengthBinning(final int binSize) {
+        // TODO: change contract to compute defaults with 0.005, 0.025, 0.975, 0.995 to have the 99% and 95%
```

Comment:
add a ticket for this -> more related with default values for the LDdecay tool. And remove TODO